pyrefly/lib/test/new_type.rs

```diff
@@ -136,6 +136,21 @@ y: type[Any] = Foo  # E: `type[Foo]` is not assignable to `type[Any]`
     "#,
 );
 
+testcase!(
+    test_new_type_type_argument,
+    r#"
+from typing import NewType
+
+Thing = NewType("Thing", int)
+ThingType = type[Thing]
```

Comment:
I'm wondering if we should also spend effort flagging invalid `type[NewType]` usages at annotation/alias definition time? Would be willing to follow up in a separate PR.

Reply:
I think if we have a `type[NewType]` we may want to just emit the error at the definition site, and infer the type as `Any`.

Then we wouldn't need a custom error everywhere it's being used, so the error at the definition site would supersede this PR.